src/main/java/com/iexec/worker/result/ResultService.java

```diff
@@ -361,6 +367,67 @@ public ComputedFile getComputedFile(String chainTaskId) {
         return computedFile;
     }
 
+    /**
+     * Write computed file. Most likely used by tee-post-compute.
+     * TODO: check compute stage is successful
+     *
+     * @param computedFile computed file to be written
+     * @return true is computed file is successfully written to disk
+     */
+    public boolean writeComputedFile(ComputedFile computedFile) {
+        if (computedFile == null || StringUtils.isEmpty(computedFile.getTaskId())) {
+            log.error("Cannot write computed file [computedFile:{}]", computedFile);
+            return false;
+        }
+        String chainTaskId = computedFile.getTaskId();
+        ChainTaskStatus chainTaskStatus =
+                iexecHubService.getChainTask(chainTaskId)
+                        .map(ChainTask::getStatus)
+                        .orElse(null);
+        if (chainTaskStatus != ChainTaskStatus.ACTIVE) {
+            log.error("Cannot write computed file if task is not active " +
+                            "[chainTaskId:{}, computedFile:{}, chainTaskStatus:{}]",
+                    chainTaskId, computedFile, chainTaskStatus);
+            return false;
+        }
+        if (new File(workerConfigService.getTaskOutputDir(chainTaskId)
+                + IexecFileHelper.SLASH_COMPUTED_JSON).exists()) {
```

Comment:
Maybe it would be useful to have a `workerConfigService.getTaskComputedJsonFilepath(chainTaskId)` (?)

Reply:
Refactored it, good eyes, thank you!